sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -174,7 +174,9 @@ function getRequestBody(body?: unknown, contentType: string = ""): RequestBody {
 
   if (ArrayBuffer.isView(body)) {
     if (body instanceof Uint8Array) {
-      return { body: binaryArrayToString(body) };
+      return firstType === "application/octet-stream"
+        ? { body }
+        : { body: binaryArrayToString(body) };
```

Comment:
Actually I'm not sure why we do the converntion `binaryArrayToString` for Uint8Array. Is it because the main assumption is application/json type?

Reply:
Yeah I will continue looking into this area.